crates/router/src/connector/paypal/transformers.rs

```diff
@@ -925,6 +946,11 @@ impl<F, T>
             capture_id: None,
             psync_flow: item.response.intent
         });
+        let purchase_units = item
+            .response
+            .purchase_units
+            .first()
+            .ok_or(errors::ConnectorError::MissingConnectorTransactionID)?;
```

Comment:
If purchase_units is empty the use id in the connector_response_reference_id as a fallback, instead of throwing error